packages/devtools_app/lib/src/shared/utils.dart

```diff
@@ -243,25 +243,40 @@ class DebounceTimer {
     Duration duration,
     FutureOr<void> Function(Timer timer) callback,
   ) {
-    _timer = Timer.periodic(duration, (timer) async {
-      // If the callback is still running, then don't run again.
-      if (_isRunning) {
-        return;
-      }
-
-      try {
-        _isRunning = true;
-        await callback(timer);
-      } finally {
-        _isRunning = false;
-      }
+    // Start a timer with no duration so that the callback will be triggered
+    // immediately.
+    _timer = Timer(const Duration(), () async {
```

Comment:
we don't need another timer for this.  just call `callback` directly once before starting the periodic timer

Reply:
Since it is a FutureOr, I couldn't find a way to trigger the callback inside the constructor, as they can't be async.

Reply:
Why does it need to be a `FutureOr`? if you change the parameter type to `Future`, you can do something like this:
```
unawaited(callback().then((_) {
  /* start periodic timer */
));
```